apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -576,28 +589,30 @@ def _create_nat_epg_for_ext_net(self, l3out_name, ext_epg_name,
                 # make external EPG use NAT contract
                 self.apic_manager.ensure_external_epg_consumed_contract(
                     l3out_name, nat_contract, external_epg=ext_epg_name,
-                    transaction=trs)
+                    owner=tenant_name, transaction=trs)
                 self.apic_manager.ensure_external_epg_provided_contract(
                     l3out_name, nat_contract, external_epg=ext_epg_name,
-                    transaction=trs)
+                    owner=tenant_name, transaction=trs)
                 # make L3-out use NAT-vrf
                 self.apic_manager.ensure_external_routed_network_created(
-                    l3out_name, context=nat_vrf_name,
+                    l3out_name, context=nat_vrf_name, owner=tenant_name,
                     transaction=trs)
 
                 # create shadow L3-out and shadow external-epg
```

Comment:
What is the expectation in a world where each tenant has his own VRF? Consider that even "shared" networks are still created in the ACI tenant corresponding to their "tenant_id", and not in the common tenant.